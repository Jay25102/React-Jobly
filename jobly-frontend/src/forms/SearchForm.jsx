import React, { useState } from "react";
import "./SearchForm.css";

/**
 * Search component used in CompanyList and JobList components.
 * When submitted, the component calls the searchFor prop which
 * is a function defined by the components that render it.
 */
function SearchForm({searchFor}) {
    const [searchTerm, setSearchterm] = useState("");
    console.debug("SearchForm", "searchFor=", typeof searchFor);

    function handleSubmit(e) {
        e.preventDefault();
        searchFor(searchTerm.trim() || undefined);
        setSearchterm(searchTerm.trim());
    }

    function handleChange(e) {
        setSearchterm(e.target.value);
    }

    return (
        <div className="SearchForm mb-4">
            <form className="form-inlinne" onSubmit={handleSubmit}>
                <input 
                    className="form-control form-control-lg flex-grow-1"
                    name="searchTerm"
                    placeholder="Enter term..."
                    value={searchTerm}
                    onChange={handleChange}
                />
                <button className="btn btn-lg btn-primary" type="submit">Submit</button>
            </form>
        </div>
    )
}

export default SearchForm;