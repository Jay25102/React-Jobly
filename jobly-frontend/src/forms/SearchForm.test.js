import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "./SearchForm";

describe("SearchForm", function() {
    test("it renders without crashing", function() {
        render(<SearchForm/>);
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(<SearchForm/>);
        expect(asFragment()).toMatchSnapshot();
    });
});