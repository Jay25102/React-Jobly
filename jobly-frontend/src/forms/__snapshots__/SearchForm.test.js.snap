// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileForm it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="searchTerm"
        placeholder="Enter term..."
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`SearchForm it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="searchTerm"
        placeholder="Enter term..."
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
